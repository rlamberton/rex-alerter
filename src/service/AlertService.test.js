import getNewAlerts from "./AlertService";

beforeEach(() => {
    jest.spyOn(global, 'fetch').mockResolvedValue({
        ok: true,
        json: jest.fn().mockResolvedValue(
            { eth: { symbol: 'ETH-BTC', quoteVolume: 20, askRate: 100, name: "Etherium", status: 'ONLINE' } ,
            'ETH-BTC': { symbol: 'ETH', status: 'ONLINE', name: "Etherium" },
            ETH: { name: 'Etherium', symbol: 'ETH', status: 'ONLINE'}}
        )
    })
});
  
afterEach(() => {
    jest.restoreAllMocks();
});

test('check snapshot', async () => {
    let results = await getNewAlerts();
    expect( results ).toMatchSnapshot();

    results = await getNewAlerts();
    expect( results ).toMatchSnapshot();

    jest.spyOn(global, 'fetch').mockResolvedValue({
        ok: true,
        json: jest.fn().mockResolvedValue(
            { eth: { symbol: 'ETH-BTC', quoteVolume: 20, askRate: 1, status: 'ONLINE' } ,
            'ETH-BTC': { symbol: 'ETH', status: 'ONLINE' }}
        )
    })

    results = await getNewAlerts();
    results[0].time = '08:30';
    expect( results ).toMatchSnapshot();
});