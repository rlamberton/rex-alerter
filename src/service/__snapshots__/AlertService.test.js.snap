// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check snapshot 1`] = `Array []`;

exports[`check snapshot 2`] = `Array []`;

exports[`check snapshot 3`] = `
Array [
  Object {
    "logoUrl": undefined,
    "name": undefined,
    "newPrice": 1,
    "oldPrice": 100,
    "pctChange": -99,
    "symbol": "ETH-BTC",
    "time": "08:30",
    "url": "https://global.bittrex.com/Market/Index?MarketName=BTC-ETH",
    "volume": 20,
  },
]
`;
